src/axom/primal/tests/primal_nurbs_curve.cpp

```diff
@@ -0,0 +1,831 @@
+      {
+        EXPECT_NEAR(p[i], p2[i], 1e-13);
+      }
+    }
+
+    if(t >= 0.66 && t <= 0.77)
+    {
+      PointType p3 = bezier_list[2].evaluate((t - 0.66) / (0.77 - 0.66));
+      for(int i = 0; i < DIM; ++i)
+      {
+        EXPECT_NEAR(p[i], p3[i], 1e-13);
+      }
+    }
+
+    if(t >= 0.77)
+    {
+      PointType p4 = bezier_list[3].evaluate((t - 0.77) / (1 - 0.77));
+      for(int i = 0; i < DIM; ++i)
+      {
+        EXPECT_NEAR(p[i], p4[i], 1e-13);
+      }
+    }
+  }
+}
+
+//------------------------------------------------------------------------------
+TEST(primal_nurbscurve, nurbs_reverse_orientation)
+{
+  SLIC_INFO("Testing NURBS reverse orientation");
+
+  const int DIM = 3;
+  using CoordType = double;
+  using PointType = primal::Point<CoordType, DIM>;
+  using NURBSCurveType = primal::NURBSCurve<CoordType, DIM>;
+
+  PointType data[4] = {PointType {0.6, 1.2, 1.0},
+                       PointType {1.3, 1.6, 1.8},
+                       PointType {2.9, 2.4, 2.3},
+                       PointType {3.2, 3.5, 3.0}};
+
+  double weights[4] = {1.0, 2.0, 3.0, 4.0};
+
+  NURBSCurveType curve(data, weights, 4, 3);
+
+  // Insert some knots to stress test reversal
+  curve.insertKnot(0.33, 3);
+  curve.insertKnot(0.66, 1);
+  curve.insertKnot(0.77, 2);
+
+  NURBSCurveType curve_reversed(curve);
+  curve_reversed.reverseOrientation();
+
+  for(double t = 0.0; t < 1.0; t += 0.05)
+  {
+    PointType p = curve.evaluate(t);
+    PointType p_reversed = curve_reversed.evaluate(1.0 - t);
+
+    for(int i = 0; i < DIM; ++i)
+    {
+      EXPECT_NEAR(p[i], p_reversed[i], 1e-13);
+    }
+  }
+}
+
+//------------------------------------------------------------------------------
+TEST(primal_nurbscurve, nurbs_winding_numbers)
+{
+  // Define a nurbs curve that represents a circle
+  const int DIM = 2;
+  using CoordType = double;
+  using PointType = primal::Point<CoordType, DIM>;
+  using NURBSCurveType = primal::NURBSCurve<CoordType, DIM>;
+
+  PointType data[7] = {PointType {1.0, 0.0},
+                       PointType {1.0, 2.0},
+                       PointType {-1.0, 2.0},
+                       PointType {-1.0, 0.0},
+                       PointType {-1.0, -2.0},
+                       PointType {1.0, -2.0},
+                       PointType {1.0, 0.0}};
+  double weights[7] = {1.0, 1. / 3., 1. / 3., 1.0, 1. / 3., 1. / 3., 1.0};
+
+  double knots[11] = {0.0, 0.0, 0.0, 0.0, 0.5, 0.5, 0.5, 1.0, 1.0, 1.0, 1.0};
+
+  NURBSCurveType circle(data, weights, 7, knots, 11);
+
+  // Check the winding number on a simple grid of points
+  for(double x = -2.0; x <= 2.0; x += 0.201)
+  {
+    for(double y = -2.0; y <= 2.0; y += 0.201)
+    {
+      PointType query {x, y};
+      double gwn = winding_number(query, circle);
+      if(x * x + y * y > 1.0)
+      {
+        EXPECT_DOUBLE_EQ(std::lround(gwn), 0.0);
+      }
+      else
+      {
+        EXPECT_DOUBLE_EQ(std::lround(gwn), 1.0);
+      }
+    }
+  }
+}
```

Comment:
Would it better to move this test to primal winding number test file?

Reply:
Makes sense. While I'm digging around in that test file, it probably makes sense to move the rational Bezier curve winding number test there too.